test/integration/replicaset/replicaset_test.go

```diff
@@ -404,6 +392,14 @@ func testScalingUsingScaleSubresource(t *testing.T, c clientset.Interface, rs *v
 	}); err != nil {
 		t.Fatalf("Failed to set .Spec.Replicas of scale subresource for rs %s: %v", rs.Name, err)
 	}
+
+	newRS, err = rsClient.Get(rs.Name, metav1.GetOptions{})
+	if err != nil {
+		t.Fatalf("Failed to obtain rs %s: %v", rs.Name, err)
+	}
+	if *newRS.Spec.Replicas != replicas {
```

Comment:
Note that it takes time for `scale.Spec.Replicas` to become 3. Pooling is needed if we want to have `*newRS.Spec.Replicas != scale.Spec.Replicas`.

Reply:
sgtm :)